pkg/cloud/rgraph/algo/trclosure/transitive_closure_test.go

```diff
@@ -0,0 +1,257 @@
+package trclosure
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/meta"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph/rnode/fake"
+	"github.com/google/go-cmp/cmp"
+	"k8s.io/klog/v2"
+)
+
+func TestTransitiveClosure(t *testing.T) {
+	// No t.Parallel() due to use of fake.Mocks.Add().
+	const project = "proj1"
+	mockCloud := cloud.NewMockGCE(nil) // TODO: project router
+
+	addNode := func(from string, toList []string, opts ...func(*fake.Builder)) *fake.Builder {
+		id := fake.ID(project, meta.GlobalKey(from))
+		ret := fake.NewBuilder(id)
+		ret.SetOwnership(rnode.OwnershipManaged)
+		ret.SetState(rnode.NodeExists)
+
+		for _, opt := range opts {
+			opt(ret)
+		}
+		for _, to := range toList {
+			ret.FakeOutRefs = append(ret.FakeOutRefs, rnode.ResourceRef{
+				From: ret.ID(),
+				To:   fake.ID(project, meta.GlobalKey(to)),
+			})
+		}
+		if fake.Mocks.Add(ret) {
+			// We panic here to catch buggy test setup early.
+			panic(fmt.Sprintf("duplicate fake.Mocks.Add(%s)", ret.ID()))
+		}
+		klog.Infof("fake.Mocks.Add(%s)", ret.ID())
+		return ret
+	}
+	addNodeToGraph := func(g *rgraph.Builder, from string, toList []string, opts ...func(*fake.Builder)) {
```

Comment:
I don't see `opts` to be ever passed here? Do you really need it?

Reply:
No, but as test code there isn't much harm having this as an option to future tests...